pkg/logging/logging.go

```diff
@@ -167,86 +153,93 @@ func init() {
 type LogOptions map[string]string
 
 // InitializeDefaultLogger returns a logrus Logger with a custom text formatter.
-func InitializeDefaultLogger() *logrus.Logger {
-	logger := logrus.New()
-	logger.Formatter = GetFormatter(DefaultLogFormat)
-	logger.SetLevel(LevelStringToLogrusLevel[DefaultLogLevelStr])
-	return logger
-}
-
-// GetLogLevelFromConfig returns the log level provided via global
-// configuration. If the logging level is invalid, ok will be false.
-func GetLogLevelFromConfig() (logrus.Level, bool) {
-	return logOptions.GetLogLevel()
+func InitializeDefaultLogger() (logger *logrus.Logger) {
+	logger = logrus.New()
+	logger.SetFormatter(GetFormatter(DefaultLogFormat))
+	logger.SetLevel(DefaultLogLevel)
+	return
 }
 
 // GetLogLevel returns the log level specified in the provided LogOptions. If
-// it is not set in the options, ok will be false.
-func (o LogOptions) GetLogLevel() (level logrus.Level, ok bool) {
-	level, ok = LevelStringToLogrusLevel[strings.ToLower(o[LevelOpt])]
+// it is not set in the options, it will return the default level.
+func (o LogOptions) GetLogLevel() (level logrus.Level) {
+	levelOpt, ok := o[LevelOpt]
+	if !ok {
+		return DefaultLogLevel
+	}
+
+	var err error
+	if level, err = logrus.ParseLevel(levelOpt); err != nil {
+		logrus.WithError(err).Error("parsing log level, using default one")
```

Comment:
Similar to other comment:
```suggestion
		logrus.WithError(err).Warning("Ignoring user-configured log level")
```

Reply:
added as part of 4bdca22ba4aae67f176e34e3863a93e7223307c7 :+1: